packages/indexer/src/sinks/sqlite.ts

```diff
@@ -101,8 +103,14 @@ export class SqliteSink<
   }
 }
 
-export const sqlite = async (args: SqliteArgs & SqliteSinkOptions) => {
-  const { filename, mode, driver, ...sinkOptions } = args;
-  const db = await open({ filename, mode, driver });
-  return new SqliteSink(db, sinkOptions);
+export const sqlite = <TData extends Record<string, unknown>>(
```

Comment:
I think this one should also accept a `Database` as argument so that users can configure it.